### ISSUE TITLE
The -s flag does not work with floats like 4., but the -e flag does

### ISSUE THREAD

**Issue by mooreryan** [soedinglab/MMseqs2#640]
## Expected Behavior

Expect the `-s` option to work with floats like `4.`.  I.e.,  `-s 4.` does not work, but `-s 4` and `-s 4.0` do work. (Note that the evalue option `-e` does work with all three: `-e 4`, `-e 4.`, and `-e 4.0`.)

## Current Behavior

It gives an error: `Error in argument -s`

## Steps to Reproduce (for bugs)

Does not work.

```
mmseqs easy-search queries.faa targets.faa out.tsv tmp -s 4.
```

Does work

```
mmseqs easy-search queries.faa targets.faa out.tsv tmp -s 4
mmseqs easy-search queries.faa targets.faa out.tsv tmp -s 4.0
mmseqs easy-search queries.faa targets.faa out.tsv tmp -e 4
mmseqs easy-search queries.faa targets.faa out.tsv tmp -e 4.
mmseqs easy-search queries.faa targets.faa out.tsv tmp -e 4.0
```

## MMseqs Output (for bugs)

```
$ mmseqs easy-search queries.faa targets.faa out.tsv tmp -s 4.

... lots of output describing the command line options ...

Error in argument -s
```

## Your Environment

* commit: 7e2840992948ee89dcc336522dc98a74fe0adf00
* statically compiled: `mmseqs-linux-avx2`
* OS: Ubuntu 22.04.1 LTS

**Comment by milot-mirdita** [comment]
The following a regex checks for valid floats: `"^[0-9]*(\\.[0-9]+)?$"`
I am not sure we want to change this though.

**Comment by milot-mirdita** [comment]
We have two different regex checking for valid floats for sensitivity (`"^[0-9]*(\\.[0-9]+)?$"`) and evalue (`"^([-+]?[0-9]*\\.?[0-9]+([eE][-+]?[0-9]+)?)|[0-9]*(\\.[0-9]+)?$"`). That inconsistent, though I am not sure there is a lot of value in allowing float values like `4.`.

**Comment by mooreryan** [comment]
Yeah I don't think it's all that valuable to allow it...I only noticed because I was using mmseqs in an automated pipeline, and by default floats are converted like that in the software I was using.  So definitely not a common issue.  Just wanted to let you know about it.
